crates/core/tedge_agent/src/agent.rs

```diff
@@ -628,21 +629,27 @@ impl SmAgent {
 }
 
 #[cfg(test)]
-fn get_restart_command() -> Command {
+fn get_restart_operation_command_vec() -> Vec<Command> {
+    let mut vec = vec![];
     // running `echo 6` with no sudo
     let mut command = std::process::Command::new(INIT_COMMAND);
     command.arg("6");
-    command
+    vec.push(command);
+    vec
```

Comment:
Minor - you can avoid a `mut` here.

```suggestion
vec![command]
```